seisbench/models/dkpn.py

```diff
@@ -0,0 +1,784 @@
+
+            if len(versions) == 0:
+                raise ValueError(f"No version for weight '{name}' available.")
+            version_str = max(versions, key=version.parse)
+
+        weight_path, metadata_path = cls._pretrained_path(name, version_str)
+
+        cls._ensure_weight_files(
+            name, version_str, weight_path, metadata_path, force, wait_for_file
+        )
+
+        if metadata_path.is_file():
+            with open(metadata_path, "r") as f:
+                weights_metadata = json.load(f)
+        else:
+            weights_metadata = {}
+        model_args = weights_metadata.get("model_args", {})
+        model_args["in_channels"] = 4
+        model = cls(**model_args)
+
+        model._weights_metadata = weights_metadata
+        model._parse_metadata()
+
+        state_dict = torch.load(weight_path)
+        old_weight = state_dict["inc.weight"]
+        state_dict["inc.weight"] = torch.zeros(
+            old_weight.shape[0], old_weight.shape[1] + 1, old_weight.shape[2]
+        ).type_as(old_weight)
+        state_dict["inc.weight"][:, :3, ...] = old_weight
+        state_dict["inc.weight"][:, 3, ...] = old_weight[:, 0, ...]
+        model.load_state_dict(state_dict)
+        return model
+
+
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================  PREPROC
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+
+
+class PreProc(object):
+    def __init__(self, **kwargs):
+        self.stream = None  # --> Instantiated everytime in the work() class-method
+        self.__dict__.update(kwargs)
+        self.eps = 1e-10
+
+    def __call__(self, state_dict):
+        """
+        Provides an augmentation for seisbench.generate.generator.GenericGenerator.add_augmentations().
+        """
+
+        labels, metadata_labels = state_dict["y"]
+        waveforms, metadata_waveforms = state_dict["X"]  # metadata is the relevant row-field of CSV
+        waveforms = self.__matrix_cfs__(waveforms)
+
+        # ================================  Emulates WINDOW-LEVEL NORM
+        # --- Remove FP stabilization
+        fstab = int(metadata_waveforms["trace_sampling_rate_hz"]*self.fp_stabilization)
```

Comment:
I'm not sure I understand these lines. I guess they should remove some padding in the beginning, but they also truncate at the end?